pulp_2_tests/tests/rpm/api_v2/test_duplicate_uploads.py

```diff
@@ -86,3 +88,97 @@ def do_test(self, feed, type_id, body, unit_key=None):
                 'unit_key': unit_key
             }, repo)
             self.assertIsNone(call_report['result'])
+
+
+class DuplicateUploadAndCopyTestCase(unittest.TestCase):
+    """Test same-name content uploads do not duplicate in `PULP_MANIFEST`."""
+
+    def test_upload_copy_manifest(self):
+        """Test same-name content uploads do not duplicate in `PULP_MANIFEST`.
+
+        Steps:
+
+        1. Create two new feed-less iso-repo repositories.
+        2. Upload content and import it into the source repository. Assert
+           the upload and import was successful.
+        3. Copy the content to a target repo. Assert the copy was successful.
+        4. Upload identically named but different content and import it into
+           the source repository. Assert the upload and import was successful.
+        5. Copy the content to the target repo.
+        6. Assert the target manifest only has one entry for the content.
+        """
+        cfg = config.get_config()
+        if cfg.pulp_version < Version('2.20'):
+            raise unittest.SkipTest('This test requires Pulp 2.20 or newer.')
+        client = api.Client(cfg, api.json_handler)
+
+        # 1. Create two iso-repo
+        repos = []
+        iso = utils.http_get(FILE_URL)
+        unit_key = {
+            'checksum': hashlib.sha256(iso).hexdigest(),
+            'name': os.path.basename(urlsplit(FILE_URL).path),
+            'size': len(iso),
+        }
+        data = {
+            'importer_type_id': 'iso_importer',
+            'notes': {'_repo-type': 'iso-repo'},
+        }
+        repos.append(client.post(REPOSITORY_PATH, gen_repo(**data)))
+        self.addCleanup(client.delete, repos[0]['_href'])
+        repos.append(client.post(REPOSITORY_PATH, gen_repo(**data)))
+        self.addCleanup(client.delete, repos[1]['_href'])
+
+        # 2. Import the units into the source repo and verify
+        call_report = upload_import_unit(cfg, iso, {
+            'unit_type_id': 'iso',
+            'unit_key': unit_key
+        }, repos[0])
+        self.assertIsNone(call_report['result'])
+
+        # Only one iso unit should exist in the source repo
+        self.assertEqual(
```

Comment:
```suggestion
        units = search_units(cfg, repos[0], {'type_ids': ['iso']})
        self.assertEqual(len(units)),1, units)
```

Reply:
I am using a similar call 3 times. If I re-use `units` each time, the value of `units` is re-used. Seems like a bad practice.

Reply:
I also don't like shadowing variables so on that situation I go with 2 options:

1. Name you shadowable variable with an `_` like `_units` this way others reading the code knows by naming convention that this variable is not to rely on because it is either private or temporary, there are some examples where temp var are named simply `_`.
2. Have different variables for each calls like `units1`, `units2`, ...

Reply:
We can update this PR once we have a different solution. My comment was a suggestion to make the PR code closer to the rest of the code basis.